### PR TITLE
feat: CI - cache lerna dependencies

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**
feat

**Did you add tests for your changes?**
Not yet

**If relevant, did you update the documentation?**
NA

**Summary**
- https://github.com/actions/cache/blob/main/examples.md#node---lerna
- Try to use cached deps to reduce CI time

**Does this PR introduce a breaking change?**
No

**Other information**

### CODE REVIEW

**Comment by anshumanv** [comment]
We're hitting cache it seems 
image

Redundant deps install time has significantly dropped 

/cc @evilebottnawi

**Comment by anshumanv** [comment]
My bad, time has not dropped even though we're hitting the cache 😞

**Comment by evilebottnawi** [comment]
@anshumanv strange, based on docs we don't hit caches on windows

**Comment by anshumanv** [comment]
yep we're not hitting on windows, I checked on ubuntu and we're hitting there but install time has not come down

**Comment by evilebottnawi** [comment]
@anshumanv yep, linux is very fast, macos and windows are slow, so we need to look at them

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -67,7 +67,17 @@ jobs:
...
+        id: cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            node_modules
+            */*/node_modules
+          key: ${{ runner.os }}-${{ hashFiles('**/yarn.lock') }}
+
       - name: Install dependencies
+        if: steps.cache.outputs.cache-hit != 'true'
```
Comment (anshumanv) [inline .github/workflows/nodejs.yml:80]: @evilebottnawi we're good now, we needed another step to skip install deps when cache is present

**Review by evilebottnawi — COMMENTED** [review]
We need improve key, because
- https://github.com/webpack/webpack-cli/pull/1883/checks?check_run_id=1216340642
- https://github.com/webpack/webpack-cli/pull/1883/checks?check_run_id=1216340623

have same key, but we install webpack@4 and webpack@5, I think it should be different

**Comment by evilebottnawi** [comment]
After this we can skip webpack installation too

**Comment by anshumanv** [comment]
> After this we can skip webpack installation too

I did then it didn't install webpack@5 and didn't run core tests, let's try improving key and see if it works

**Comment by anshumanv** [comment]
@evilebottnawi updated key to have webpack version, now it skips webpack installation too and runs all tests fine

**Review by rishabh3112 — APPROVED** [review]

**Review by snitin315 — APPROVED** [review]
Nice job!

**Review by evilebottnawi — APPROVED** [review]

**Pull request merged**